File: solutions/javascript/eliuds-eggs/1/eliuds-eggs.js
//
// This is only a SKELETON file for the 'Eliud's Eggs' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const eggCount = (displayValue) => {
  let number = 0;

  while (displayValue > 1) {
    const digit = displayValue & 1;
    displayValue = displayValue >> 1;
    number += digit;
  }
  number += displayValue;
  return number;
};

File: solutions/java/eliuds-eggs/1/src/main/java/EliudsEggs.java
public class EliudsEggs {
    public int eggCount(int number) {
        int output = 0;
        while (number > 1) {
            int digit = number & 1;
            number = number >> 1;
            output += digit;
        }
        output += number;
        return output; 
    }
}
